webui/react/src/pages/ExperimentDetails/ExperimentSingleTrialTabs.tsx

```diff
@@ -125,21 +130,22 @@ const ExperimentSingleTrialTabs: React.FC<Props> = (
   }, [ trialDetails, stopPolling ]);
 
   useEffect(() => {
-    if (trialId != null) return;
-    return () => stopPollingFirstTrialId();
-  }, [ trialId, stopPollingFirstTrialId ]);
+    if (trialId === undefined && wontHaveTrials === false) return;
```

Comment:
fix: shouldn't this be `trialId !== undefined && wontHaveTrials === false`? Or maybe more simply `trialId && !wontHaveTrials`.

Reply:
aah I was a bit confused around that check 🤔 let me see

Reply:
does [this](https://github.com/determined-ai/determined/pull/2772/commits/71483c8e975eba610eaa7f8d5e8d76f92aa9b2f9) make sense?